BDArmory/Guidances/MissileGuidance.cs

```diff
@@ -254,18 +254,23 @@ public static Vector3 GetAirToAirFireSolution(MissileBase missile, Vessel target
             float leadTime = 0;
             float targetDistance = Vector3.Distance(targetVessel.transform.position, missile.transform.position);
 
-            Vector3 simMissileVel = 500 * (targetPosition - missile.transform.position).normalized;
+            //Vector3 simMissileVel = 500 * (targetPosition - missile.transform.position).normalized;
 
             MissileLauncher launcher = missile as MissileLauncher;
+            /*
             float optSpeed = 400; //TODO: Add parameter
             if (launcher != null)
             {
-                optSpeed = launcher.optimumAirspeed;
+                optSpeed = launcher.optimumAirspeed; //so it assumes missiles start out immediately possessing all their velocity instead of having to accelerate? That explains alot.
             }
             simMissileVel = optSpeed * (targetPosition - missile.transform.position).normalized;
 
             leadTime = targetDistance / (float)(targetVessel.Velocity() - simMissileVel).magnitude;
             leadTime = Mathf.Clamp(leadTime, 0f, 8f);
+            */
+            float accel = launcher.thrust / missile.part.mass;
+            leadTime = ((float)Math.Sqrt(accel * (accel + (8 * targetDistance))) - accel) / (2 * accel) - (Time.fixedDeltaTime * 1.5f); //quadratic equation for accel to find time from known force and vel
```

Comment:
What's the quadratic equation you're solving here?
This formula looks a bit odd. E.g., where does the factor 8 come from, the `(... - accel) / (2 * accel)` simplifies to give a constant factor of `-1/2` for some reason and why the `1.5*Δt` factor?

Reply:
Inverse accel formula; distance D is Accel * Time, if Distance and Accel are known, find Time. Might not be quadratic, but that's what's listed in the comment under the torpedo drop time calc, which is where I grabbed it from. 
The 8 looks to be a conversion to convert the standard (-a+√a^2-4bc)/2a into a 2 variable (ax & dx = t) equation instead of 3 (ax/bx/cx = x)
Doing some mathing out I can confirm that it spits out the correct time to reach D dist with A accel.
Shouldn't be subtracting that 0.03s at the end, though, that'll cause an off-by-a-frame aim error.

Reply:
That doesn't seem right. The relevant kinematic equation here is `d = v*t + 1/2*a*t^2`, which gives
```
t = (-v ± sqrt(v^2 - 4*(a/2)*(-d))) / 2*(a/2)
  = (-v ± sqrt(v^2 + 2*a*d)) / a
```
If `a` and `d` are the same sign (`-9.8m/s^2` and `-altitude` for the torpedo drop), then the `-`ve branch of the sqrt is always the correct one. The torpedo drop on line 1128 of MissileLauncher.cs also appears to assume that `v=0`, which would be reasonable for a real plane, but probably not for BDA.
If `a` and `d` are opposite sign, then the `v^2 + 2*a*d` needs to be checked for sign to avoid a math domain error and some logic is needed for which branch of the sqrt to take (or careful formulation of the problem such that `a` and `d` are the same sign).

For the lead time on missiles, `v` and `a` should be the relative velocity and acceleration between the missile and the target plane (these can be projected onto the direction between them to simplify the problem to 1d). (Ignoring the target's acceleration other than the common gravity both experience simplifies things a bit in the following.) However, to complicate things, unlike dropping the torpedo, the missile has a top speed (`V`), so if the distance to the target is beyond the distance required for the missile to get up to speed, then beyond that the lead time would be linear. E.g.,
```
v = launch speed, V = optimum speed, a = missile acceleration
Δv = relative launch speed, ΔV = relative optimum speed, assume relative acceleration = a
T = (V - v) / a = time to get up to optimum speed
D = Δv * T + 1/2 * a * T^2 = relative distance to get up to optimum speed
d = Δv * t + 1/2 * a * t^2, for d < D ⇒ solve quadratic equation for t
  = D + ΔV * (t - T), for d > D ⇒ solve linear equation for t
So, t = (-Δv - sqrt(Δv^2 + 2*a*d)) / a, for d < D
      = (D - d) / ΔV + T, otherwise
```

Reply:
``a`` and ``d`` should always be the same sign, no? Distance and accel are both going to be > 0 (unless missile has negative thrust in the config, but that's not our problem).
Technically speaking, accel is also dependent on boost time and boost thrust, then cruise thrust and cruise time, where the current calc is assuming just boost time for the max 8 sec the calc is extending out; optimalVel (assuming .cfg value is correct) should probably keep things about where they should be though.
Good point about the torp drop vel; it's assuming level flight, and not dive bombing or similar. Switch that over as well?
That said, it might be simpler in the torp's case to replace the drop delay with a simple has-it-hit-the-water-yet``FlightGlobals.getAltitudeAtPos(part.transform.position) < 0`` check in the ``BoostRoutine()``.

Reply:
The current implementation for the torpedo was to delay the drop time until it hit the water, so the torpedo wouldn't start the boost coroutine and have thrust while still airborne, but a ``yield return new WaitWhileFixed(() => vessel.splashed);`` or similar would work just as well.

Reply:
Fair enough. I'd do
```csharp
yield return new WaitUntilFixed(() => vessel == null || vessel.LandedOrSplashed);
if (vessel == null || !vessel.Splashed) abort...
```
though.

Reply:
> ``d = Δv * t + 1/2 * a * t^2, for d < D ⇒ solve quadratic equation for t``   

To return to your earlier comment about finding intercept time, where is ``t`` getting defined for this? It can't be from after ``t`` is found in the resulting ``t = (-Δv - sqrt(Δv^2 + 2*a*d)) / a`` to then do a check for if the linear equation should be used instead, because the quadratic requires a defined ``d`` to solve for ``t``?

Reply:
`d` is known, it's the distance to the target (technically, it's the distance in the direction of the target, but assuming that the missile can turn to face the target pretty quickly, you can take it as simply the distance). Then you're solving that for `t` if `d < D`.